File: core/src/main/java/com/mcreater/amclcore/model/oauth/MinecraftRequestModel.java
package com.mcreater.amclcore.model.oauth;

import com.google.gson.annotations.SerializedName;
import com.mcreater.amclcore.annotations.OAuthLoginModel;
import com.mcreater.amclcore.annotations.RequestModel;
import lombok.Builder;
import lombok.Data;

import java.util.List;
import java.util.Map;
import java.util.UUID;

@Data
@Builder
@OAuthLoginModel
@RequestModel
public class MinecraftRequestModel {
    private UUID username;
    private List<Object> roles;
    private Map<String, Object> metadata;
    @SerializedName("access_token")
    private String accessToken;
    @SerializedName("expires_in")
    private int expiresIn;
    @SerializedName("token_type")
    private String tokenType;
}

File: core/src/main/java/com/mcreater/amclcore/model/oauth/MinecraftRequestModel.kt
package com.mcreater.amclcore.model.oauth

import com.google.gson.annotations.SerializedName
import com.mcreater.amclcore.annotations.RequestModel
import java.util.*

@RequestModel
data class MinecraftRequestModel(
    var username: UUID? = null,
    var roles: List<Any>? = null,
    var metadata: Map<String, Any>? = null,
    @SerializedName("access_token") var accessToken: String? = null,
    @SerializedName("expires_in") var expiresIn: Int = 0,
    @SerializedName("token_type") var tokenType: String? = null,
)
